homeassistant/components/go2rtc/__init__.py

```diff
@@ -46,22 +53,47 @@
 )
 
 
-async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
-    """Set up WebRTC from a config entry."""
-    if binary := entry.data.get(CONF_BINARY):
+CONFIG_SCHEMA = vol.Schema({DOMAIN: {vol.Optional(CONF_URL): cv.url}})
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up WebRTC."""
+    url: str | None = None
+    if DOMAIN not in config or not (url := config[DOMAIN].get(CONF_URL)):
+        if not is_docker_env():
+            _LOGGER.warning("Go2rtc URL required in non-docker installs")
+            return False
+        if not (binary := await _get_binary(hass)):
+            _LOGGER.error("Could not find go2rtc docker binary")
+            return False
+
         # HA will manage the binary
         server = Server(hass, binary)
-
-        entry.async_on_unload(server.stop)
         await server.start()
 
-    client = Go2RtcClient(async_get_clientsession(hass), entry.data[CONF_URL])
+    if url is not None:
+        # Validate user provided URL
+        try:
+            client = Go2RtcClient(async_get_clientsession(hass), url)
+            await client.streams.list()
+        except Exception:  # noqa: BLE001
+            _LOGGER.warning("Could not connect to go2rtc instance on %s", url)
+            return False
+    else:
+        url = "http://localhost:1984/"
+
+    client = Go2RtcClient(async_get_clientsession(hass), url)
 
     provider = WebRTCProvider(client)
-    entry.async_on_unload(async_register_webrtc_provider(hass, provider))
+    async_register_webrtc_provider(hass, provider)
```

Comment:
Should we also deregister on shutdown too?

Reply:
That seems pointless?